### ISSUE TITLE
Single-frame move routes shouldn't block the playable character

### ISSUE THREAD

**Issue by Zegeri** [EasyRPG/Player#626]
Create a parallel event that forces a move route to the hero each frame. This move route must be completable in a single frame (with frequency 8 and without Move Up/Down/Left/Right commands). Run the game.

Result in RPG_RT: Can move the hero around
Result in EasyRPG: Can't move the hero

[Here's a test sample.](https://dl.dropboxusercontent.com/u/30277047/HeroWithMoveRoute.7z)

0e2d19e60 might be a fix for this, but I still haven't checked if it breaks other things (very likely).

**Comment by fdelapena** [comment]
Related game issue (for cross-reference):
https://easy-rpg.org/forums/viewtopic.php?f=4&t=605

**Comment by Ghabry** [comment]
This also breaks Sacred Tears: TRUE.
Each frame the Player executed custom move route "Set character graphic".

**Comment by Ghabry** [comment]
And your suggested fix fixes it :+1:

**Comment by Ghabry** [comment]
Similiar one #579 
Moved them now to milestone 0.4.1
I vote for applying the fix after 0.4, then we have some time for finding regressions.

**Comment by Zegeri** [comment]
I already found some regressions. With the patch, the player will skip touch events if he doesn't stop moving. There will be probably more bugs. Maybe the dynrpg plugin will shed some light on this.

Here's an uglier patch safe from regressions.

```diff
diff --git a/src/game_player.cpp b/src/game_player.cpp
index 8d235e0..5534822 100644
--- a/src/game_player.cpp
+++ b/src/game_player.cpp
@@ -583,7 +583,7 @@ bool Game_Player::GetOffVehicle() {
 bool Game_Player::IsMovable() const {
 	if (IsMoving() || IsJumping())
 		return false;
-	if (IsMoveRouteOverwritten())
+	if (IsBlockedByMouveRoute())
 		return false;
 	if (location.boarding || location.unboarding)
 		return false;
@@ -594,6 +594,23 @@ bool Game_Player::IsMovable() const {
     return true;
 }
 
+bool Game_Player::IsBlockedByMouveRoute() const {
+	if (!IsMoveRouteOverwritten() || GetMoveRouteIndex() > 0)
+		return false;
+
+	// Check if it includes a blocking move command
+	for (const auto& move_command : GetMoveRoute().move_commands) {
+		int code = move_command.command_id;
+		if ((code <= RPG::MoveCommand::Code::move_forward) || // Move
+			(code <= RPG::MoveCommand::Code::face_away_from_hero && GetMoveFrequency() < 8) || // Turn
+			(code == RPG::MoveCommand::Code::wait) || // Wait
+			(code <= RPG::MoveCommand::Code::end_jump)) // Jump
+				return true;
+	}
+
+	return false;
+}
+
 bool Game_Player::InVehicle() const {
 	return location.vehicle > 0;
 }
diff --git a/src/game_player.h b/src/game_player.h
index 0a95df1..788cd3f 100644
--- a/src/game_player.h
+++ b/src/game_player.h
@@ -102,6 +102,7 @@ public:
 
 	bool GetOnOffVehicle();
 	bool IsMovable() const;
+	bool IsBlockedByMouveRoute() const;
 	bool InVehicle() const;
 	bool InAirship() const;
 	Game_Vehicle* GetVehicle() const;
```
